torchci/clickhouse_queries/queue_times_historical/query.sql

```diff
@@ -0,0 +1,28 @@
+SELECT
+    formatDateTime(
+            CASE
+                WHEN {granularity:String} = 'minute' THEN toStartOfMinute(q.time)
+                WHEN {granularity:String} = 'hour' THEN toStartOfHour(q.time)
+                WHEN {granularity:String} = 'day' THEN toStartOfDay(q.time)
+                WHEN {granularity:String} = 'week' THEN toStartOfWeek(q.time)
+                WHEN {granularity:String} = 'month' THEN toStartOfMonth(q.time)
+                WHEN {granularity:String} = 'year' THEN toStartOfYear(q.time)
+                ELSE toStartOfDay(q.time)  -- Default to day if granularity is not recognized
+                END,
+            '%Y-%m-%dT%H:%i:%s'
+    ) AS granularity_bucket,
```

Comment:
Curious as to why use formatDateTime opposed to date_trunc?  I also wonder if we need to be worried about timezones here?

Reply:
Both good questions.

I'll see if this could be simplified with `date_trunc`.